experimental/algorithm/LAGraph_RegularPathQuery.c

```diff
@@ -0,0 +1,357 @@
+    GrB_Index states = ns ;              // # pairs in the current
+                                         // correspondence between the graph and
+                                         // the NFA
+
+    GrB_Index rows = 0 ;                 // utility matrix row count
+    GrB_Index cols = 0 ;                 // utility matrix column count
+    GrB_Index vals = 0 ;                 // utility matrix value count
+
+    GrB_Matrix *A = NULL ;
+    GrB_Matrix *B = NULL ;
+    GrB_Matrix *BT = NULL ;
+
+    LG_ASSERT (reachable != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (G != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (R != NULL, GrB_NULL_POINTER) ;
+    LG_ASSERT (S != NULL, GrB_NULL_POINTER) ;
+
+    (*reachable) = NULL ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (G[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (G[i], msg)) ;
+    }
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL) continue ;
+        LG_TRY (LAGraph_CheckGraph (R[i], msg)) ;
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &A, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (G[i] == NULL) 
+        {
+            A[i] = NULL ;
+            continue ;
+        }
+
+        A[i] = G[i]->A;
+    }
+
+    LG_TRY (LAGraph_Malloc ((void **) &B, nl, sizeof (GrB_Matrix), msg)) ;
+    LG_TRY (LAGraph_Malloc ((void **) &BT, nl, sizeof (GrB_Matrix), msg)) ;
+
+    for (int32_t i = 0; i < nl; i++)
+    {
+        if (R[i] == NULL)
+        {
+            B[i] = NULL ;
+            continue ;
+        }
+
+        B[i] = R[i]->A;
+        if (R[i]->kind == LAGraph_ADJACENCY_UNDIRECTED ||
+            R[i]->is_symmetric_structure == LAGraph_TRUE)
+        {
+            BT[i] = B[i] ;
+        }
+        else
+        {
+            // BT[i] could be NULL and the matrix will be transposed by a
+            // descriptor
+            BT[i] = R[i]->AT ;
+        }
+    }
+
+    for (int i = 0; i < nl; i++) 
+    {
+        if (A[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&ng, A[i])) ;
+        break ;
+    }
+
+    for (int i = 0; i < nl; i++) 
+    {
+        if (B[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&nr, B[i])) ;
+        break ;
+    }
+
+    // Check all the matrices in graph adjacency matrix decomposition are
+    // square and of the same dimensions
+    for (int i = 0; i < nl; i++)
+    {
+        if (A[i] == NULL) continue ;
+
+        GRB_TRY (GrB_Matrix_nrows (&rows, A[i])) ;
+        GRB_TRY (GrB_Matrix_ncols (&cols, A[i])) ;
+
+        LG_ASSERT_MSG (rows == ng && cols == ng, LAGRAPH_NOT_CACHED,
+            "all the matrices in the graph adjacency matrix decomposition "
+            "should have the same dimensions and be square") ;
+    }
+
+    // Check all the matrices in NFA adjacency matrix decomposition are
+    // square and of the same dimensions
+    for (int i = 0; i < nl; i++)
```

Comment:
Use consistent types in the loop everywhere.

Thanks @vkutuev for noticing it.

Reply:
Now using `size_t` everywhere. @vkutuev, please, resolve if you're ok with it.

Reply:
Closing, since discussed verbally Vladimir Aleksandrovich is ok.